transport/split/writer.go

```diff
@@ -18,9 +18,18 @@ import (
 	"io"
 )
 
+// repeatedSplit represents a split sequence of count blocks with bytes length.
+type repeatedSplit struct {
+	count int
+	bytes int64
+}
+
 type splitWriter struct {
-	writer      io.Writer
-	prefixBytes int64
+	writer io.Writer
+	// Bytes until the next split. This must always be > 0, unless splits are done.
+	nextSplitBytes int64
```

Comment:
is this field needed? you could directly check `remainingSplits[0]` and it might save some code maintaining additional counter

Reply:
It was needed because remainingSplits[0] may have count > 1, and you want to edit the bytes just for the first split, and preserve the original value for subsequent splits.